sdcm/nemesis.py

```diff
@@ -1523,6 +1523,79 @@ def disrupt_network_block(self):
             self.target_node.traffic_control(None)
 
+    def disrupt_nodetool_removenode(self):
+        """
+        https://docs.scylladb.com/operating-scylla/procedures/cluster-management/remove_node/
+
+        1.terminate
+        2.Full repair
+        3.Nodetool removenode
+        4.Sleep 10 minutes
+        5.Add new node
+        """
+
+        self._set_current_disruption('TerminateAndRemoveNodeMonkey')
+        node_to_remove = self.target_node
+        up_normal_nodes = self.cluster.get_nodes_up_and_normal(verification_node=node_to_remove)
+        node = random.choice(up_normal_nodes)
+        # node_to_remove must be different than node
+        while node is node_to_remove:
+            node = random.choice(up_normal_nodes)
+
+        # node_to_remove must not be the only seed in cluster
+        num_of_seed_nodes = len(self.cluster.seed_nodes)
+        if node_to_remove.is_seed and num_of_seed_nodes < 2:
+            raise UnsupportedNemesis("To running seed decommission the cluster must contains at least 2 seed nodes")
+
+        # get node's host_id
+        removed_node_status = self.cluster.get_node_status_dictionary(
+            ip_address=node_to_remove.ip_address, verification_node=node)
+        assert removed_node_status is not None, "failed to get host_id using nodetool status"
+        host_id = removed_node_status["host_id"]
+
+        # node stop and make sure its "DN"
+        node_to_remove.stop_scylla_server(verify_up=True, verify_down=True)
+
+        # full cluster repair
+        for node in up_normal_nodes:
+            if node is not node_to_remove:
+                self.repair_nodetool_repair(node=node)
+
+        # nodetool removenode 'host_id'
+        self.log.info("Removing node with the following host_id: {}".format(host_id))
+        node.run_nodetool("removenode {}".format(host_id), ignore_status=False, verbose=True)
+
+        # sleep and verify node is removed by nodetool status
+        time.sleep(10*60)
+        removed_node_status = self.cluster.get_node_status_dictionary(
+            ip_address=node_to_remove.ip_address, verification_node=node)
+        assert not node_to_remove.db_up, f'Scylla db is still up, therefore the node was not removed'
+        assert removed_node_status is None, "Node was not removed properly (Node status:{})".format(removed_node_status)
+
+        self._terminate_cluster_node(node_to_remove)
+        # add new node
+        new_node = self._add_and_init_new_cluster_node()
+        # in case the removed node was a seed
+        if node_to_remove.is_seed:
+            new_node.is_seed = True
+            seed_address = ",".join([node.ip_address for node in self.cluster.seed_nodes])
+            seed_provider = [dict(class_name='org.apache.cassandra.locator.SimpleSeedProvider',
+                                  parameters=[dict(seeds=seed_address)])]
+            for node in self.cluster.nodes:
+                node.patch_scylla_yaml(seed_provider=seed_provider)
```

Comment:
Is this enough to add node as a seed?
Don't we need to do something special as part of "add_and_init_new_cluster_node"?
Is the bootstrap for them the same?

Reply:
yes this is the procedure, you boot it normally, and then promote it to be seed node.
if we have this logic now in now places, it might be a function on BaseCluster. and not copied from place to place

Reply:
I will add it to BaseCluster